app/sampl/base_settings.py

```diff
@@ -100,17 +101,43 @@
 
 CRISPY_TEMPLATE_PACK = "bootstrap4"
 
-LOGIN_TO_AWS = False
+# container engine is either "docker" or "singularity"
+CONTAINER_ENGINE = "docker"
+
 
+ECR_BASE_URL = os.environ["ECR_BASE_URL"]
+ECR_SAMPLLEAGUE_URL = os.environ["ECR_SAMPLLEAGUE_URL"]
 
-def run_aws_login():
-    subprocess.run(
-        [
-            "aws ecr get-login-password --region us-east-2 --profile sampl_pull | docker login --username AWS --password-stdin 103125031445.dkr.ecr.us-east-2.amazonaws.com"
-        ],
-        shell=True,
-        check=True,
-    )
 
+def run_aws_login(container_engine):
+    if container_engine == "docker":
+        subprocess.run(
+            [
+                f"aws ecr get-login-password --region us-east-2 --profile sampl_pull | docker login --username AWS --password-stdin {ECR_BASE_URL}"
+            ],
+            shell=True,
+            check=True,
+        )
+    elif container_engine == "singularity":
+        os.environ["SINGULARITY_DOCKER_USERNAME"] = "AWS"
+        aws_call = subprocess.run(
+            ["aws ecr get-login-password --region us-east-2 --profile sampl_pull"],
```

Comment:
```suggestion
            ["aws", "ecr", "get-login-password", "--region", "us-east-2", "--profile", "sampl_pull"],
```
and see next comment